pkg/delete/delete_loader.go

```diff
@@ -71,7 +71,7 @@ func LoadEntriesToDelete(fs afero.Fs, deleteFile string) (DeleteEntries, error)
 	context := &loaderContext{
 		fs:         fs,
 		deleteFile: filepath.Clean(deleteFile),
-		knownApis:  api.NewAPIs(),
+		knownApis:  api.NewAPIs().Filter(api.RemoveDisabled),
```

Comment:
This single line the 'real' fix, right? I'm okay adding refactoring changes to a PR, but as you already started, it would make it easier for me to review if things are nicely split into separate commits for refactoring and the change.